src/java/main/org/apache/zookeeper/server/NettyServerCnxnFactory.java

```diff
@@ -109,6 +110,20 @@ public void channelConnected(ChannelHandlerContext ctx,
                     zkServer, NettyServerCnxnFactory.this);
             ctx.setAttachment(cnxn);
 
+            InetAddress addr = ((InetSocketAddress)cnxn.channel.getRemoteAddress()).getAddress();
+            Set<NettyServerCnxn> s = ipMap.get(addr);
```

Comment:
So I can't imagine this causing much more than the occasional extra single connection, but I do think it is worth fixing.

My feeling here is that if we move the ipMap logic to the factory this should be easier to fix as synchronization in a singleton is much simpler than trying to figure out what Netty does with these CnxnChannelHandler objects.